activation/wire/wire_v2_test.go

```diff
@@ -1,17 +1,65 @@
 package wire
 
 import (
-	"encoding/binary"
 	"math/rand/v2"
 	"testing"
 
 	fuzz "github.com/google/gofuzz"
-	"github.com/spacemeshos/merkle-tree"
 	"github.com/stretchr/testify/require"
 
 	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/signing"
 )
 
+type testAtxV2Opt func(*ActivationTxV2)
+
+func WithPublishEpoch(epoch types.EpochID) testAtxV2Opt {
+	return func(atx *ActivationTxV2) {
+		atx.PublishEpoch = epoch
+	}
+}
+
+func WithMarriageCertificate(sig *signing.EdSigner, refAtx types.ATXID, atxPublisher types.NodeID) testAtxV2Opt {
+	return func(atx *ActivationTxV2) {
+		certificate := MarriageCertificate{
+			ReferenceAtx: refAtx,
+			Signature:    sig.Sign(signing.MARRIAGE, atxPublisher.Bytes()),
+		}
+		atx.Marriages = append(atx.Marriages, certificate)
+	}
+}
+
+func newActivationTxV2(opts ...testAtxV2Opt) *ActivationTxV2 {
```

Comment:
Weirdly, the option functions are exported when `newActivationTxV2` is not. I think they shouldn't be because they are not usable.